akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -0,0 +1,368 @@
+    phash: Digest,
+    chash: Digest,
+}
+
+/// Generate N audit proofs from a new tree
+async fn generate_audit_proofs(n: usize) -> Result<Vec<AuditInformation>, akd::errors::AkdError> {
+    let db = AsyncInMemoryDatabase::new();
+    let vrf = HardCodedAkdVRF {};
+    let akd = Directory::<_, _>::new::<Hasher>(&db, &vrf, false).await?;
+    let mut proofs = vec![];
+    // gather the hash + azks for epoch "0" (init)
+    let mut azks = akd.retrieve_current_azks().await?;
+    let mut phash = akd.get_root_hash::<Hasher>(&azks).await?;
+
+    for _epoch in 1..=n {
+        // generate (n) epochs of updates on the same user
+        let t_value = format!("certificate{}", _epoch);
+        akd.publish::<Hasher>(vec![(
+            AkdLabel::from_utf8_str("user"),
+            AkdValue::from_utf8_str(&t_value),
+        )])
+        .await?;
+        // generate the append-only proof
+        let proof = akd
+            .audit::<Hasher>((_epoch - 1) as u64, _epoch as u64)
+            .await?;
+        // update the azks + retrieve the "next" hash
+        azks = akd.retrieve_current_azks().await?;
+        let chash = akd.get_root_hash::<Hasher>(&azks).await?;
+
+        // we have everything we need, pass it along
+        let info = AuditInformation {
+            phash,
+            chash,
+            proof,
+        };
+        proofs.push(info);
+
+        // reset the "previous" hash for the next iteration
+        phash = chash;
+    }
+    Ok(proofs)
+}
+
+/// Flush the storage bucket of all blobs + delete the bucket (if it exists)
+async fn maybe_flush_storage(shared_config: &SdkConfig, bucket_name: &str) -> Result<()> {
+    // Build the S3 config from the shared SdkConfig
+    let config = S3ConfigBuilder::from(shared_config)
+        .retry_config(RetryConfig::disabled())
+        .build();
+    // get the S3 client
+    let client = Client::from_conf(config);
+    // check if the bucket exists
+    let buckets = client.list_buckets().send().await?;
+
+    if let Some(bucket_names) = buckets.buckets() {
+        log::info!("Found {} buckets in the test storage", bucket_names.len());
+
+        for bucket in bucket_names {
+            if let Some(maybe_name) = bucket.name() {
+                if maybe_name == bucket_name {
+                    log::info!("Found target bucket {}, deleting.", bucket_name);
+
+                    // From: https://docs.aws.amazon.com/sdk-for-rust/latest/dg/rust_s3_code_examples.html
+                    let objects = client.list_objects_v2().bucket(bucket_name).send().await?;
+                    let mut delete_objects: Vec<ObjectIdentifier> = vec![];
+                    for obj in objects.contents().unwrap_or_default() {
+                        let obj_id = ObjectIdentifier::builder()
+                            .set_key(Some(obj.key().unwrap().to_string()))
+                            .build();
+                        delete_objects.push(obj_id);
+                    }
```

Comment:
Can you do something like below to make this slightly more succinct? 
```rust
let delete_objects = objects.contents().unwrap_or_default().iter()
    .map(...builder_stuff)
    .collect();
```

Reply:
that was originally copy-pasta from the AWS docs, but this is cleaner for sure